dev_apps/music-nga/test/marionette/lib/music.js

```diff
@@ -0,0 +1,493 @@
+  },
+
+  get albumsTab() {
+    return this.client.helper.waitForElement(Music.Selector.albumsTab);
+  },
+
+  get songsTab() {
+    return this.client.helper.waitForElement(Music.Selector.songsTab);
+  },
+
+  // get the first song from the active frame.
+  get firstSong() {
+    return this.client.helper.waitForElement(Music.Selector.firstSong);
+  },
+
+  // Helper for the getter.
+  _getListItemsData: function(frame) {
+    assert.ok(frame, 'Frame must be valid.' + frame);
+
+    this.client.switchToFrame(frame);
+
+    var listItems = this.client.executeScript(function () {
+      var list = document.getElementById('list');
+      var elementsData = [];
+      var elements = list.querySelectorAll('li');
+      for(var i = 0; i < elements.length; i++) {
+        var data = {};
+        var a = elements[i].getElementsByTagName('a');
+        if (a.length) {
+          data.filePath = a[0].dataset.filePath;
+          data.href = a[0].href;
+        }
+        var h3 = elements[i].getElementsByTagName('h3');
+        if (h3.length) {
+          data.title = h3[0].textContent;
+        }
+        var p = elements[i].getElementsByTagName('p');
+        if (p.length) {
+          data.text = p[0].textContent;
+        }
+        elementsData.push(data);
+      }
+      return elementsData;
+    });
+    this.switchToMe();
+    return listItems;
+  },
+
+  get firstListItem() {
+    return this.client.helper.waitForElement(Music.Selector.firstListItem);
+  },
+
+  get artistsListItemsData() {
+    return this._getListItemsData(this.artistsViewFrame);
+  },
+
+  get songsListItemsData() {
+    return this._getListItemsData(this.songsViewFrame);
+  },
+
+  get songs() {
+    return this._getListItems(Music.Selector.viewsSublist);
+  },
+
+  get playButton() {
+    return this.client.findElement(Music.Selector.playButton);
+  },
+
+  get shareButton() {
+    return this.client.findElement(Music.Selector.shareButton);
+  },
+
+  get title() {
+    var header = this.client.findElement(Music.Selector.header);
+    return header.findElement(Music.Selector.titleText);
+  },
+
+  get sublistShuffleButton() {
+    return this.client.helper.waitForElement(
+      Music.Selector.sublistShuffleButton);
+  },
+
+  // TODO(gareth): Move this shareMenu stuff into the helper.
+  get shareMenu() {
+    // Switch to the system app first.
+    this.client.switchToFrame();
+    return this.client.helper.waitForElement(Music.Selector.shareMenu);
+  },
+
+  get progressBar() {
+    return this.client.findElement(Music.Selector.progressBar);
+  },
+
+  get pickDoneButton() {
+    return this.client.findElement(Music.Selector.pickDoneButton);
+  },
+
+  get header() {
+    return this.client.findElement(Music.Selector.header);
+  },
+
+  get playerIcon() {
+    return this.client.findElement(Music.Selector.playerIcon);
+  },
+
+  get isPlaying() {
+    return this.playButton.getAttribute('class').indexOf('is-pause') === -1;
```

Comment:
This doesn't look correct. However, the `<music-controls>` web component exposes a `paused` attribute that you can test.